components/scream/src/share/io/scorpio_input.cpp

```diff
@@ -33,10 +33,11 @@ AtmosphereInput (const ekat::Comm& comm,
                  const ekat::ParameterList& params,
                  const std::shared_ptr<const grid_type>& grid,
                  const std::map<std::string,view_1d_host>& host_views_1d,
-                 const std::map<std::string,FieldLayout>&  layouts)
+                 const std::map<std::string,FieldLayout>&  layouts,
+                 const std::map<std::string, std::string>& dtypes)
```

Comment:
This first draft of dtype feature is probably not going to work, and will need some revision: you are passing views with a Real data type, so anyt value of dtype different from "Real" would be erroneous. Given that, this is not really an "added feature".

I would hold off any mod that has to do with dtype, until we have a better design in mind. This PR should focus only on the time index feature.